peer-prep/src/Components/QuestionList/Question.jsx

```diff
@@ -0,0 +1,103 @@
+import React, {useState} from 'react'
+import './Question.css'
+import delete_icon from '../Assets/bin.png'
+import edit_icon from '../Assets/pencil.png'
+
+export const Question = ({question, i, deleteQuestion, updateQuestion}) => {
+    const [selected, setSelected] = useState(null);
+    const [title, setTitle] = useState(question.title);
+    const [difficulty, setDifficulty] = useState(question.complexity);
+    const [topic, setTopic] = useState(question.category);
+    const [description, setDescription] = useState(question.description)
+
+    const toggle = (i) => {
+        if (!isEdit) {
+            if(selected === i) {
+                return setSelected(null)
+            }
+            setSelected(i)
+        }
+    };
+    const [isEdit, setEdit] = useState(false);
+    let tagClass = "q-tag";
+
+    if(question.complexity.toLowerCase() === "easy") {
+        tagClass += " q-tag-green"
+    } else if (question.complexity.toLowerCase() === "medium") {
+        tagClass += " q-tag-orange"
+    } else if (question.complexity.toLowerCase() === "hard"){
+        tagClass += " q-tag-red"
+    } else {
+        tagClass += " q-tag-white"
+    }
+  return (
+    <div className= {isEdit ? "form-container": "question-container"}>
+        <div className={isEdit ? "container": "question"}>
+            <div className="q-header" onClick = {() => toggle(i)}>
+                <div className="q-name">
+                    <div className={isEdit ? "q-form-id": "q-id"}>#{i + 1}</div>
+                    {isEdit === true? <input type="text" className="q-form-title q-form-input" defaultValue = {question.title}
+                    onChange = {(e) => {setTitle(e.target.value)}}/> : 
+                    <div className={selected === i ? "q-title-show": "q-title"}>{question.title}</div>}
+                </div>
+                <span> {selected === i ? "-" : "+"}</span>
+            </div>
+            
+            {!isEdit 
+                ? 
+                <div className= {selected === i ? "q-content-show": "q-content"}>
+                    <div className="q-tag-container">
+                        <div className={isEdit ? "q-form-tags" : "q-tags"}>
+                            {isEdit === true? <input type="text" className="q-form-tag q-form-input" defaultValue = {question.complexity} 
```

Comment:
I feel like in the future we could make the question complexity a dropdown list or radio button (or really anything similar) just so that admins are limited to the three options easy, medium or hard. Likewise, I think we could also consider having a character limit on titles and categories just for the sake of avoiding overly long titles or categories. (This should probably be extended to tags as well)

Reply:
Yup, i agree with your suggestion. I'll consider adding a dropdown list in a future iteration (for the reasons you've mentioned) and I think the character limit can be helpful for the visual side of things!